common/CMakeLists.txt

```diff
@@ -20,6 +22,12 @@ target_sources(common_path PRIVATE path/platform/c++17/path.cpp)
 target_compile_options(common_path PRIVATE ${PLATFORM_SPECIFIC_COMPILE_OPTIONS})
 target_include_directories(common_path PRIVATE ${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR})
 
+check_cxx_source_compiles("#include <filesystem>
+  int main() { std::filesystem::path p(\"/\"); }" STD_FILESYSTEM_AVAILABLE)
```

Comment:
Would [`CheckSymbolExists()`](https://cmake.org/cmake/help/latest/module/CheckSymbolExists.html) work here?

Reply:
It claims that it can be unreliable when the symbol has more than one overload, and they recommend using `CheckCXXSourceCompiles`. [[1]](https://cmake.org/cmake/help/latest/module/CheckCXXSymbolExists.html#command:check_cxx_symbol_exists)